mlir/lib/Dialect/Mesh/IR/MeshOps.cpp

```diff
@@ -280,10 +244,36 @@ LogicalResult verifyMeshSymbolUses(Op op, SymbolTableCollection &symbolTable) {
   return success();
 }
 
+template <typename It>
+bool isUnique(It begin, It end) {
+  if (begin == end) {
+    return true;
+  }
+  It next = std::next(begin);
+  if (next == end) {
+    return true;
+  }
+  for (; next != end; ++next, ++begin) {
+    if (*begin == *next) {
+      return false;
+    }
+  }
+  return true;
+}
+
+LogicalResult verifyMeshAxes(Location loc, ArrayRef<MeshAxis> axes) {
+  SmallVector<MeshAxis> sorted = llvm::to_vector(axes);
+  std::sort(sorted.begin(), sorted.end());
```

Comment:
I'd like to suggest to check `std::set(axes.begin(), axes.end()).size() == axes.size()` instead so we don't need to sort axes. And the code will be more concise.

Reply:
Usually construction of `std::set` would be more expensive then doing a copy + sort + linear search for consecutive duplicate elements. I have not measured it, but my intuition is that for small numbers like < 10 this would be true.

Reply:
Make sense, but I'd like suggest to use `std::unique(begin, end) == end` instead of implementing `isUnique` ourselves.

Reply:
I would even speculate that for all size a vector would be faster.

Reply:
OK, this is fine to me.